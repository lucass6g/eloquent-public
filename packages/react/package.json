{
  "name": "@eloquent/react",
  "version": "0.0.1",
  "description": "",
  "main": "./dist/index.js",
  "files": [
    "dist"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    ".": {
      "import": "./dist/index.js",
      "require": "./dist/index.js",
      "types": "./dist/index.d.ts"
    }
  },
  "scripts": {
    "build": "tsup"
  },
  "keywords": [],
  "author": "",
  "license": "MIT",
  "devDependencies": {
    "tsup": "^8.0.2",
    "@types/react": "^18.2.66",
    "@types/react-dom": "^18.2.22",
    "react": "^18.2.0",
    "typescript": "^5.4.2"
  }
}
