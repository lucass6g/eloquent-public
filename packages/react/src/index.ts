export {Switch, Case, Default} from './components/conditional-switch'