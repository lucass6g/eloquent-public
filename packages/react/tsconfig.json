{
  "extends": "../../packages/config/base.tsconfig.json",
  "include": ["src"],
  "exclude": ["dist", "build", "node_modules"],
  "compilerOptions": {
    "forceConsistentCasingInFileNames": false,
    "rootDir": "src",
    "baseUrl": ".",
    "paths": {
      "@/*": ["./src/*"]
    }
  }
}
