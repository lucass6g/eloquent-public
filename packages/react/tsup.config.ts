import { defineConfig } from 'tsup';

export default defineConfig({
	name: "@eloquent/react",
	entry: ["src/index.ts"],
	outDir: "dist",
	target: "es2022",
	format: ["cjs", "esm"],
	splitting: false,
	sourcemap: true,
	clean: true,
	dts: true,
	external: ["react", 'react-dom', 'react/jsx-runtime'],
	cjsInterop: true,
	treeshake: true,
})