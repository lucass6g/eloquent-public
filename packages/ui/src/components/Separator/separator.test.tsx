import { render } from "@testing-library/react";
import { Separator } from ".";

describe("Separator", () => {
    describe("should match the snapshot", () => {
        it("horizontal", () => {
            const component = render(
                <div className="flex flex-col gap-2">
                    <h1>title</h1>
                    <Separator orientation="horizontal" />
                    <p>text</p>
                </div>
            );

            expect(component.baseElement).toMatchSnapshot();
        });

        it("vertical", () => {
            const component = render(
                <div className="flex h-5 items-center space-x-4 text-sm">
                    <div>test 1</div>
                    <Separator orientation="vertical" />
                    <div>test 2</div>
                </div>
            );

            expect(component.baseElement).toMatchSnapshot();
        });
    });
});
