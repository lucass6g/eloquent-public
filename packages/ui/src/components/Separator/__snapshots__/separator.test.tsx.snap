// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Separator > should match the snapshot > horizontal 1`] = `
<body>
  <div>
    <div
      class="flex flex-col gap-2"
    >
      <h1>
        title
      </h1>
      <div
        class="shrink-0 bg-border h-[1px] w-full"
        data-orientation="horizontal"
        role="none"
      />
      <p>
        text
      </p>
    </div>
  </div>
</body>
`;

exports[`Separator > should match the snapshot > vertical 1`] = `
<body>
  <div>
    <div
      class="flex h-5 items-center space-x-4 text-sm"
    >
      <div>
        test 1
      </div>
      <div
        class="shrink-0 bg-border h-full w-[1px]"
        data-orientation="vertical"
        role="none"
      />
      <div>
        test 2
      </div>
    </div>
  </div>
</body>
`;
